images/airflow/2.9.2/python/mwaa/entrypoint.py

```diff
@@ -190,15 +190,31 @@ def create_queue() -> None:
 
 
 def _requirements_has_constraints(requirements_file: str):
-    with open(requirements_file, "r") as file:
-        for line in file:
-            # Notice that this regex check will also match lines with commented out
-            # constraints flag. This is intentional as a mechanism for users who
-            # want to avoid enforcing the default Airflow constraints, yet don't want
-            # to provide a constraints file.
-            if re.search(r"-c |--constraint ", line):
-                return True
+    encodings_to_try = ["utf-8", "ascii", "latin1"]
+    for encoding in encodings_to_try:
+        try:
+            with open(requirements_file, "r", encoding=encoding) as file:
+                for line in file:
+                    # Notice that this regex check will also match lines with commented
+                    # out constraints flag. This is intentional as a mechanism for users
+                    # who want to avoid enforcing the default Airflow constraints, yet
+                    # don't want to provide a constraints file.
+                    if re.search(r"-c |--constraint ", line):
+                        return True
+            return False
+        except UnicodeDecodeError:
+            logging.warning(
+                f"Failed to decode {requirements_file} using {encoding} encoding."
+            )
 
+    # latin1 is not expected to fail, since it just reads the file as bytes (regardless
+    # of the content), so this block is not expected to be reach, but adding it to be
+    # on the safe side.
+    logging.warning(
+        f"Failed to decode {requirements_file} using {encodings_to_try} encodings. "
+        "Cannot determine whether the requirements file has constraints or not; "
+        "assuming it doesn't."
```

Comment:
if it failed to encode there is no need to add "assuming it doesn't". 
I believe following error message is enough. 
```
Failed to decode {requirements_file} using {encodings_to_try} encodings. "
```

Reply:
Why do you want to remove that part? We were unable to tell if the customer has constraints specified or not, so we ended up making an ** assumption ** that the customer requirements file doesn't have constraints, and this could be incorrect, hence we need to state that clearly in the log to avoid confusion.

Reply:
I see what you mean but I feel like "assuming" file context wording is wrong when file decoding failed. we do not know the content of the files so assuming that it doesn't contain constraints is not ideal. 
maybe change of wording will be better.

Reply:
We should just rephrase it "Cannot determine whether the requirements file has constraints or not;" "proceeding without constraining requirements" 
"assuming" is a strange way to say it